pkg/virtctl/imageupload/imageupload.go

```diff
@@ -386,6 +390,22 @@ func (c *command) run(args []string) error {
 		return err
 	}
 
+	if dataSource {
+		ds, err := virtClient.CdiClient().CdiV1beta1().DataSources(namespace).Get(context.Background(), name, metav1.GetOptions{})
```

Comment:
can you please move this block to new function? The run function has already more than 100 lines, which is too long

Reply:
I'll move it to a new function called `handleDataSource(...)`.